tests/test_fact_manager.py

```diff
@@ -0,0 +1,439 @@
+    """
+    mfd_list method must return a list.
+    """
+    fm = test_fm_fx
+
+    assert isinstance(await fm.mfd_list(), list)
+
+
+@pytest.mark.asyncio
+async def test_fm_destroy(test_fm_fx):
+    """
+    Verify _destroy function works properly.
+    """
+    fm = test_fm_fx
+
+    await fm._destroy('test', 'ru')
+
+
+@pytest.mark.asyncio
+async def test_edit_message_bad_fact(test_fm_fx):
+    """
+    Verify attempt to edit a fact that does not exist throws exception.
+    """
+    fm = test_fm_fx
+
+    with pytest.raises(ValueError):
+        await fm.edit_message('notafact', 'nope', 'Shatt', 'This is not a fact')
+
+
+@pytest.mark.asyncio
+async def test_edit_message(test_fm_fx):
+    """
+    Test edit_message functionally works.
+    """
+    fm = test_fm_fx
+    await fm.edit_message('test', 'en', 'Shatt', 'This fact has been edited during testing.')
+
+    result = await fm.find('test', 'en')
+
+    assert result.message == 'This fact has been edited during testing.'
+
+
+@pytest.mark.asyncio
+async def test_edit_message_failure(test_fm_fx):
+    """
+    Verify an error is raised if an edit on a non-existent fact is attempted.
+    """
+    fm = test_fm_fx
+
+    with pytest.raises(ValueError):
+        await fm.edit_message('nopeavi', 'nope', 'Kovacs', 'This is not a fact')
+
+
+@pytest.mark.asyncio
+async def test_fact_exists(test_fm_fx):
+    """
+    Verify a fact looked up with the exists method returns true.
+    """
+    fm = test_fm_fx
+    assert bool(await fm.exists('stats', 'en'))
+
+
+@pytest.mark.asyncio
+async def test_fact_exists_no(test_fm_fx):
+    """
+    Verify a fact looked up with the exist method returns false if not found.
+    """
+    fm = test_fm_fx
+    assert not bool(await fm.exists('asdf', 'asdfa'))
+
+
+@pytest.mark.asyncio
+async def test_facthistory_return(test_fm_fx):
+    """
+    Verify fact history returns log entries.
+    """
+    fm = test_fm_fx
+
+    # Add some log entries.
+    await fm.log('stats', 'en', 'Shatt', 'Added')
+    await fm.log('stats', 'en', 'Shatt', 'Deleted')
+    await fm.log('stats', 'en', 'Shatt', 'Added')
+    await fm.log('stats', 'en', 'Shatt', 'Marked for Delete')
+
+    history = await fm.facthistory('stats', 'en')
+    assert history is not None
+
+
+@pytest.mark.asyncio
+async def test_facthistory_invalid_query(test_fm_fx):
+    """
+    Verify fact history raises no errors on an invalid fact lookup.
+    """
+    fm = test_fm_fx
+
+    history = await fm.facthistory('s205gw', 'sdjn3nw')
+
+
+@pytest.mark.asyncio
+async def test_fact_find_return(test_fm_fx):
+    """
+    Verify fact finder returns a complete fact object.
+    """
+    fm = test_fm_fx
+    found_fact = await fm.find('stats', 'en')
+
+    assert found_fact.complete is True
+
+
+@pytest.mark.asyncio
+async def test_exist_exception_handling(test_fm_fx, monkeypatch):
+    """
+    Verify the error is caught if psycopg2.ProgrammingError is raised.
+    """
+
+    async def boomstick(*args, **kwargs):
```

Comment:
Duplicated helper method, consider refactoring into an `AsyncCallableMock` fixture?

Reply:
Mock is unit test.

Reply:
We have our own definition, under `tests/mock_callables.py`.

Reply:
i don't believe our callable mocks have the ability to raise exceptions

Reply:
AFAIK, monkeypatch's scope is the same as the parent def. when you add a helper method in this fashion.

Reply:
Our Mocks can raise exceptions, using `CallableMock.exception_to_raise`.